### PR TITLE
fix(app/js) load and render data

### PR DESCRIPTION
**Pull request by taka-sho**

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -8,3 +8,4 @@
 [libs]
 
 [options]
+suppress_comment= \\(.\\|\n\\)*\\flow-disable-line
```
Comment (9renpoto) [inline .flowconfig]: do not use this options.

File: app/entrypoint/options.jsx [hunk 2]
```diff
@@ -4,4 +4,6 @@ import { h, render } from 'preact'
 import Options from '../components/options'
 import { get } from '../actions/options'
 
-render('app', <Options {...get()} />)
+  ;(async () => {
+  render(<Options configs={await get()} />, document.body)
```
Comment (9renpoto) [inline app/entrypoint/options.jsx]: I think,  this change is bad commit

File: app/entrypoint/popup.jsx [hunk 3]
```diff
@@ -2,5 +2,10 @@
 /** @jsx h */
 import { h, render } from 'preact'
 import Popup from '../components/popup'
+import { get } from '../actions/popup'
 
-render('app', <Popup />)
+window.addEventListener('load', e => {
+  ;(async () => {
+    render(<Popup data={(await get()).data} />, document.getElementById('info'))
```
Comment (9renpoto) [inline app/entrypoint/popup.jsx]: and too.

File: package.json [hunk 4]
```diff
@@ -43,6 +43,7 @@
     "mock-local-storage": "^1.0.4",
     "npm-run-all": "^4.0.2",
     "prettier-eslint-cli": "^4.1.1",
+    "pug-cli": "^1.0.0-alpha6",
```
Comment (9renpoto) [inline package.json]: You probably do not need to use pug. Because html can not be as many as 5 lines

File: __tests__/background.spec.js [hunk 5]
```diff
@@ -29,8 +29,8 @@ describe('background', () => {
   })
 
   test('toggle', () => {
-    expect(bg.toggle()).toBe('active')
```
Comment (9renpoto) [inline __tests__/background.spec.js]: this test is true. 
first state is `off`, next state `on`

File: app/actions/background.js [hunk 6]
```diff
@@ -5,14 +5,15 @@ declare var chrome: any
 
 export default class Background {
   constructor () {
+    this.set(IS_ACTIVE, 'active')
```
Comment (9renpoto) [inline app/actions/background.js]: why first state `on` ?
please `off` state start

File: app/actions/contentscript.js [hunk 7]
```diff
@@ -18,27 +19,61 @@ declare var chrome: any
 export default class Provider {
   constructor () {
     window.addEventListener('load', async e => {
+      this.listen()
```
Comment (9renpoto) [inline app/actions/contentscript.js]: why first listen, Even if you turn it off this way you do extra things

File: app/actions/options.js [hunk 8]
```diff
@@ -16,7 +18,6 @@ export async function set ({ env, host, id, ignores, isActive }: any) {
   bg.set('USERDIVEId', id)
   bg.set('USERDIVEIgnore', ignores)
 
-  thenChrome.tabs.getCurrent(tab => {
-    thenChrome.tabs.remove(tab.id)
-  })
+  const tab = await thenChrome.tabs.getCurrent()
+  thenChrome.tabs.remove(tab.id)
```
Comment (9renpoto) [inline app/actions/options.js]: 👍

File: app/components/popup.jsx [hunk 9]
```diff
@@ -1,20 +1,44 @@
...
 type Props = {
   id: string
 }
 
 export default class Popup extends Component<Props> {
+  constructor (props: Object) {
+    super(props)
+    this.state = props
+    ;(async () => {
+      this.setState({ isActive: await isActive() })
```
Comment (9renpoto) [inline app/components/popup.jsx]: typeof `isActivate` equal `boolean`, I think.

File: app/tag.js [hunk 10]
```diff
@@ -1,6 +1,8 @@
 /* eslint no-template-curly-in-string: 0 eslint-comments/no-unlimited-disable: 1 */
 'use strict'
 ;(function (global, document, element, state) {
+  element = document.getElementById('${elementId}')
+  element.setAttribute('${attrName}', JSON.stringify({ status: 'Loading' }))
```
Comment (9renpoto) [inline app/tag.js]: where is this `state` use?

**Comment by 9renpoto** [comment]
Need more tests.

File: app/tag.js (hunk 10 above)
Comment (taka-sho) [reply app/tag.js]: the state has three statuses: `Loading` , `Blocked` and `Failed`

https://github.com/uncovertruth/userdive-tag-paster/pull/143/files#diff-378a5288126f16880a1d45efb628a424R59

File: app/actions/contentscript.js [hunk 11]
```diff
@@ -21,24 +22,67 @@ export default class Provider {
       const config = await getConfig()
       if (!config.isActive) {
         renderBadge('OFF')
+        this.listen()
```
Comment (9renpoto) [inline app/actions/contentscript.js]: why listen when `isActivate === false`

File: app/components/options.jsx [hunk 12]
```diff
@@ -31,16 +32,16 @@ export default class Options extends Component<Store, Store> {
     this.setState({ env: target.value })
   }
   onChangeHost ({ target }: any) {
-    this.setState({ env: target.value })
+    this.setState({ host: target.value })
   }
-  onChangeIgnores ({ target }: any) {
-    this.setState({ ignores: target.value })
+  onChangeignore ({ target }: any) {
```
Comment (shoota) [inline app/components/options.jsx]: should be named by use camelcase

File: app/actions/contentscript.js [hunk 13]
```diff
@@ -21,24 +22,67 @@ export default class Provider {
...
+    const state = await this.loadState()
+    renderBadge(state.pageId || '?')
+  }
   async loadState (): Object {
     const { isActive } = await getConfig()
     if (!isActive) {
-      return {}
+      return {
+        status: 'OFF',
+        message: 'to enable paster, plase click buton in popup window'
```
Comment (heavenshell) [inline app/actions/contentscript.js]: typo

File: jest.config.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (9renpoto) [inline jest.config.js]: this settings written at package.json

File: app/entrypoint/options.jsx [hunk 15]
```diff
@@ -3,5 +3,7 @@
 import { h, render } from 'preact'
 import Options from '../components/options'
 import { get } from '../actions/options'
-
-render('app', <Options {...get()} />)
+  ;(async () => {
+  const data = await get()
+  render(<Options {...data} />, document.body)
```
Comment (9renpoto) [inline app/entrypoint/options.jsx]: It is nonsense to rewrite all the body. Let's rewrite the single element

**Pull request merged**